### PR TITLE
new plugin: RescueTime Activities

### PR DESCRIPTION
**Pull request by stared**
I just developed a new plugin, and I would be excited to share it here.

**RescueTime Activities**: Monitor your RescueTime activities and distractions directly from the macOS menu bar with this xbar plugin.

Source is at https://github.com/stared/xbar-rescuetime-activities.

Screenshot 2024-11-14 at 13 21 52

This plugin is inspired by the [RescueTime plugin](https://xbarapp.com/docs/plugins/Dev/rescuetime.1h.py.html) by [Paul Traylor](https://github.com/kfdm).
I rewrote it to display a detailed list of activities rather than just a summary of productive and non-distractive time.

### CODE REVIEW

File: Time/rescuetime-activities.3m.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+# Put the API key in ~/Library/RescueTime.com/api.key
+
+import os
+import json
+import datetime
+import urllib.parse
+import urllib.request
+
+
+API_KEY_PATH = os.path.expanduser("~/Library/RescueTime.com/api.key")
```
Comment (sprak3000) [inline Time/rescuetime-activities.3m.py:23]: Consider using [xbar variables](https://xbarapp.com/docs/2021/03/14/variables-in-xbar.html) to manage this. Would make download and setup easier for others.

File: Time/rescuetime-activities.3m.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env PYTHONIOENCODING=UTF-8 python3
```
Comment (sprak3000) [inline Time/rescuetime-activities.3m.py:1]: This change allows python to be picked up from a variety of places -- macOS install, Homebrew on Intel silicon install, Homebrew on M1 silicon install, etc.

```suggestion
#!/usr/bin/env -S PATH="${PATH}:/opt/homebrew/bin:/usr/local/bin" PYTHONIOENCODING=UTF-8 python3
```

File: Time/rescuetime-activities.3m.py (hunk 1 above)
Comment (stared) [reply Time/rescuetime-activities.3m.py]: Thank you - I modified that.

File: Time/rescuetime-activities.3m.py (hunk 2 above)
Comment (stared) [reply Time/rescuetime-activities.3m.py]: Changed this line slightly (to prioritize custom Python installations).

**Comment by stared** [comment]
Thank you, @sprak3000, for the fruitful remarks - I implemented them.

**Comment by sprak3000** [comment]
@stared Thank you for your contribution!

**Comment by stared** [comment]
@sprak3000 My pleasure!
Do you know when it will become visible in "Browse plugins" in the app or on the website?

**Comment by sprak3000** [comment]
@stared 

That's up to @matryer and his schedule. It's a manual process I believe, and he runs it periodically.

**Pull request merged**